From 4a8126d863bd7c1291302da7e6b68d06cf44674c Mon Sep 17 00:00:00 2001
From: Amanda Anganes <aanganes@mitre.org>
Date: Wed, 14 Mar 2012 15:38:36 -0400
Subject: [PATCH] Fixing config/AuthorizationServerBeanDefinitionParser.java

---
 .../AuthorizationServerBeanDefinitionParser.java   |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java b/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java
index 74a004d..9d2ff06 100644
--- a/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java
+++ b/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java
@@ -112,7 +112,8 @@ public class AuthorizationServerBeanDefinitionParser extends ProviderBeanDefinit
 		if (authorizationCodeElement != null
 				&& !"true".equalsIgnoreCase(authorizationCodeElement.getAttribute("disabled"))) {
 			// authorization code grant configuration.
-			String authorizationCodeServices = authorizationCodeElement.getAttribute("services-ref");
+			//AANGANES 3/9/2011: added changed "services-ref" to "authorization-code-services-ref"
+			String authorizationCodeServices = authorizationCodeElement.getAttribute("authorization-code-services-ref");
 			String clientTokenCacheRef = authorizationCodeElement.getAttribute("client-token-cache-ref");
 
 			BeanDefinitionBuilder authorizationCodeTokenGranterBean = BeanDefinitionBuilder
-- 
1.7.3.1.msysgit.0

