From c58240f05e58360f4404bf8d9fadfeb624cc5849 Mon Sep 17 00:00:00 2001
From: Amanda Anganes <aanganes@mitre.org>
Date: Tue, 13 Mar 2012 16:40:00 -0400
Subject: [PATCH] Line 89: changed "services-ref" to "authorization-code-services-ref"

---
 .../AuthorizationServerBeanDefinitionParser.java   |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java b/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java
index 83c25e2..d1f9445 100644
--- a/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java
+++ b/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/AuthorizationServerBeanDefinitionParser.java
@@ -86,7 +86,8 @@ public class AuthorizationServerBeanDefinitionParser extends ProviderBeanDefinit
 			String approvalPage = authorizationCodeElement.getAttribute("user-approval-page");
 			String approvalHandlerRef = authorizationCodeElement.getAttribute("approval-handler-ref");
 			String approvalParameter = authorizationCodeElement.getAttribute("approval-parameter-name");
-			String authorizationCodeServices = authorizationCodeElement.getAttribute("services-ref");
+			//AANGANES 3/9/2011: added changed "services-ref" to "authorization-code-services-ref"
+			String authorizationCodeServices = authorizationCodeElement.getAttribute("authorization-code-services-ref");
 			String redirectResolverRef = authorizationCodeElement.getAttribute("redirect-resolver-ref");
 			String clientTokenCacheRef = authorizationCodeElement.getAttribute("client-token-cache-ref");
 			String authorizationCodeRedirectStrategyRef = authorizationCodeElement
-- 
1.7.3.1.msysgit.0

